app/src/main/java/com/keylesspalace/tusky/json/GuardedAdapter.kt

```diff
@@ -35,10 +35,11 @@ class GuardedAdapter<T> private constructor(
 
     override fun fromJson(reader: JsonReader): T? {
         return try {
-            delegate.fromJson(reader)
+            delegate.fromJson(reader.peekJson())
```

Comment:
I recommend you close the peeked reader after usage so the Okio buffer segments can move back to the pool:
`reader.peekJson().use { delegate.fromJson(it) }`